ironfish-cli/src/commands/wallet/chainport/send.ts

```diff
@@ -0,0 +1,421 @@
+      transaction: RawTransactionSerde.serialize(rawTransaction).toString('hex'),
+      account: from,
+    })
+
+    const bytes = Buffer.from(postTransaction.content.transaction, 'hex')
+    const transaction = new Transaction(bytes)
+    const hash = transaction.hash().toString('hex')
+
+    if (postTransaction.content.accepted === false) {
+      this.warn(`Transaction '${hash}' was not accepted into the mempool`)
+    }
+
+    const transactionUrl = getExplorer(networkId)?.getTransactionUrl(hash)
+
+    if (transactionUrl) {
+      this.log(`\nBlock explorer: ${transactionUrl}`)
+    }
+
+    if (flags.watch) {
+      this.log('')
+
+      await watchTransaction({
+        client,
+        logger: this.logger,
+        account: from,
+        hash,
+      })
+    } else {
+      this.log(`Run ironfish wallet:transaction ${hash} to get the status of your transaction.`)
+      this.log(
+        `Run ironfish wallet:transaction:watch ${hash} to wait for the status of your transaction to be confirmed/ expired.`,
+      )
+    }
+  }
+
+  private async getAndValidateInputs(client: RpcClient, networkId: number) {
+    const { flags } = await this.parse(BridgeCommand)
+
+    let from = flags.account?.trim()
+    let to = flags.to?.trim()
+    let assetId = flags.assetId
+
+    if (!from) {
+      const response = await client.wallet.getDefaultAccount()
+
+      if (!response.content.account) {
+        this.error(
+          `No account is currently active.
+            Use ironfish wallet:create <name> to first create an account`,
+        )
+      }
+
+      from = response.content.account.name
+    }
+
+    if (!to) {
+      to = await CliUx.ux.prompt('Enter the public address of the recipient', {
+        required: true,
+      })
+    }
+
+    if (!validator.isAddress(to)) {
+      this.error('Invalid to ethereum address')
+    }
+
+    if (flags.expiration !== undefined && flags.expiration < 0) {
+      this.error('Expiration sequence must be non-negative')
+    }
+
+    const tokens = await fetchChainportVerifiedTokens(networkId)
+
+    if (assetId == null) {
+      const asset = await selectAsset(client, from, {
+        action: 'send',
+        showNativeAsset: true,
+        showNonCreatorAsset: true,
+        showSingleAssetChoice: false,
+        filter: (asset) => {
+          return tokens.some((t) => t.web3_address === asset.id)
+        },
+      })
+
+      assetId = asset?.id
+
+      if (!assetId) {
+        assetId = Asset.nativeId().toString('hex')
+      }
+    }
+
+    const asset: ChainportVerifiedToken | undefined = tokens.find(
+      (t) => t.web3_address === assetId,
+    )
+
+    if (!asset) {
+      const names = tokens.map(
+        (t, index) => `${index + 1}. ${t.name} (${t.symbol}) - ${t.web3_address}`,
+      )
+
+      this.error(
+        `Asset ${assetId} not supported by Chainport. Here are the supported tokens: \n\n${names.join(
+          '\n',
+        )}\n`,
+      )
+    }
```

Comment:
Do we need this check now that we pass the filter to `selectAsset`?

Reply:
Yes, this is still useful in case the user uses `--assetId` and passes an id for an asset not supported by chainport